src/data_access/ConversationDataAccessObject.java

```diff
@@ -30,6 +31,17 @@ public PromptGenerator getPromptGenerator() {
 
     @Override
     public void save(PromptGenerator promptGenerator) {
+        ConversationHistory conversationHistory = promptGenerator.getConversationHistory();
+
+        // Compress old conversation before saving.
+        String conversationToCompress = conversationHistory.getConversationToCompress();
+        String prompt = "Summarize the following so that it can be used on ChatGPT for context.\n"
+                + conversationToCompress;
+        if (conversationToCompress != null) {
+            DummyCompressionChatGPTAPI dummyCompressionChatGPTAPI = new DummyCompressionChatGPTAPI();
+            String compressedConversation = dummyCompressionChatGPTAPI.getResponse(prompt);
+            conversationHistory.addCompressedConversation(compressedConversation);
+        }
         this.promptGenerator = promptGenerator;
     }
```

Comment:
Nice I'll approve regardless but this can easily be swapped out for GPT4DataAccess object?

Reply:
Yup I was thinking of using GPT3DataAccess object but same idea